tests/test_healthcheck.py

```diff
@@ -128,29 +130,40 @@ def test_check_fails_if_at_least_check_for_one_item_fails(self):
 
 
 class TestFilesExistHealthCheck(TestCase):
-    @patch('os.path.isfile')
-    def test_ok_if_all_files_exist(self, is_file_mock):
-        is_file_mock.return_value = True
-        check = FilesExistHealthCheck(('file1', 'file2'), check_id='checkid')
+    def test_ok_if_all_files_exist(self):
+        tmpfile1 = NamedTemporaryFile()
+        tmpfile2 = NamedTemporaryFile()
+        file1 = tmpfile1.name
+        file2 = tmpfile2.name
+        check = FilesExistHealthCheck((file1, file2), check_id='checkid')
         check.run()
         self.assertTrue(check.is_ok)
-        self.assertEqual(check.details, {'file1': 'exists',
-                                         'file2': 'exists'})
+        self.assertEqual(check.details, {file1: 'exists',
+                                         file2: 'exists'})
 
-    @patch('os.path.isfile')
-    def test_ok_if_at_least_one_file_doesnt_exist(self, is_file_mock):
-        is_file_mock.side_effect = lambda fn: False if fn == 'file2' else True
-        check = FilesExistHealthCheck(('file1', 'file2'), check_id='checkid')
+    def test_not_ok_if_at_least_one_file_doesnt_exist(self):
+        tmpfile1 = NamedTemporaryFile()
+        file1 = tmpfile1.name
+        check = FilesExistHealthCheck((file1, 'file2'), check_id='checkid')
         check.run()
         self.assertFalse(check.is_ok)
-        self.assertEqual(check.details, {'file1': 'exists',
+        self.assertEqual(check.details, {file1: 'exists',
                                          'file2': 'NO SUCH FILE'})
 
+    @patch('os.stat')
+    def test_ok_if_file_exists_with_wrong_permissions(self, stat_mock):
+        stat_mock.side_effect = OSError(errno.EACCES, 'Permission denied')
+        self.test_ok_if_all_files_exist()
```

Comment:
I don't want to save characters, I want to reuse code intelligently so that if something needs to change it causes less of a cascade of required changes in other places in the code. What you're suggesting is increasing the amount of logic and complexity in the code, which is error-prone. Case in point, your suggestion is wrong. `.run()` doesn't return anything, it sets the `ok` attribute which needs to be checked. This is also testing the wrong thing because the result of the check is supposed to be true since the files do exist but have the wrong permissions. It also doesn't confirm that the `details` are correct, which other tests have handled.

What I've written is that:

    test_ok_if_file_exists_with_wrong_permissions

is the same as

test_ok_if_all_files_exist

except that `os.stat` has been patched to show that permission errors don't make a difference.